commons/che-core-commons-gwt/src/main/java/org/eclipse/che/ide/util/IDProvider.java

```diff
@@ -0,0 +1,25 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2015 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.ide.util;
+
+/**
+ * Provides ID for the object.
+ *
+ * @author Vlad Zhukovskiy
+ */
+public interface IDProvider {
```

Comment:
This is a proposal to unify mechanism of providing debug id for the gwt components based on their data objects